rest-api/src/main/java/com/cloudbees/workflow/rest/external/FlowNodeExt.java

```diff
@@ -167,10 +167,10 @@ public void addBasicNodeData(FlowNode node, String execNodeName, ExecDuration du
         setId(node.getId());
         setName(node.getDisplayName());
         setExecNode(execNodeName);
-        if (Stapler.getCurrentRequest() != null) {
+        if (get_links() == null) {
```

Comment:
It was more a safety thing to make sure it's not overwriting any links that are already set but you're right not required.